integration/proxy/teleterm_test.go

```diff
@@ -26,67 +27,97 @@ import (
-		databaseURI := uri.NewClusterURI(rootClusterName).
+		profileName, _, err := net.SplitHostPort(pack.Root.Cluster.Web)
+		require.NoError(t, err)
+
+		databaseURI := uri.NewClusterURI(profileName).
 			AppendDB(pack.Root.MysqlService.Name)
 
-		testGatewayCertRenewal(t, pack, "", creds, databaseURI)
+		testDBGatewayCertRenewal(t, pack, "", databaseURI)
 	})
 	t.Run("leaf cluster", func(t *testing.T) {
+		profileName, _, err := net.SplitHostPort(pack.Root.Cluster.Web)
+		require.NoError(t, err)
+
 		leafClusterName := pack.Leaf.Cluster.Secrets.SiteName
-		databaseURI := uri.NewClusterURI(rootClusterName).
+		databaseURI := uri.NewClusterURI(profileName).
 			AppendLeafCluster(leafClusterName).
 			AppendDB(pack.Leaf.MysqlService.Name)
 
-		testGatewayCertRenewal(t, pack, "", creds, databaseURI)
+		testDBGatewayCertRenewal(t, pack, "", databaseURI)
 	})
 	t.Run("ALPN connection upgrade", func(t *testing.T) {
 		// Make a mock ALB which points to the Teleport Proxy Service. Then
 		// ALPN local proxies will point to this ALB instead.
 		albProxy := helpers.MustStartMockALBProxy(t, pack.Root.Cluster.Web)
 
-		databaseURI := uri.NewClusterURI(rootClusterName).
+		// Note that profile name is taken from tc.WebProxyAddr. Use
+		// albProxy.Addr() as profile name in case it's different from
+		// pack.Root.Cluster.Web (e.g. 127.0.0.1 vs localhost).
+		profileName, _, err := net.SplitHostPort(albProxy.Addr().String())
+		require.NoError(t, err)
+
+		databaseURI := uri.NewClusterURI(profileName).
 			AppendDB(pack.Root.MysqlService.Name)
 
-		testGatewayCertRenewal(t, pack, albProxy.Addr().String(), creds, databaseURI)
+		testDBGatewayCertRenewal(t, pack, albProxy.Addr().String(), databaseURI)
 	})
 }
-func testGatewayCertRenewal(t *testing.T, pack *dbhelpers.DatabasePack, albAddr string, creds *helpers.UserCreds, databaseURI uri.ResourceURI) {
-	tc, err := pack.Root.Cluster.NewClientWithCreds(helpers.ClientConfig{
-		Login:   pack.Root.User.GetName(),
-		Cluster: pack.Root.Cluster.Secrets.SiteName,
+
+func testDBGatewayCertRenewal(t *testing.T, pack *dbhelpers.DatabasePack, albAddr string, databaseURI uri.ResourceURI) {
+	t.Helper()
+
+	testGatewayCertRenewal(
+		t,
+		pack.Root.Cluster,
+		pack.Root.User.GetName(),
+		albAddr,
+		daemon.CreateGatewayParams{
+			TargetURI:  databaseURI.String(),
+			TargetUser: pack.Root.User.GetName(),
+		},
+		mustConnectDatabaseGateway,
+	)
+}
+
+type testGatewayConnectionFunc func(*testing.T, gateway.Gateway)
+
+func testGatewayCertRenewal(t *testing.T, inst *helpers.TeleInstance, username, albAddr string, params daemon.CreateGatewayParams, testConnection testGatewayConnectionFunc) {
+	t.Helper()
+
+	tc, err := inst.NewClient(helpers.ClientConfig{
+		Login:   username,
+		Cluster: inst.Secrets.SiteName,
 		ALBAddr: albAddr,
-	}, *creds)
+	})
 	require.NoError(t, err)
 	// Save the profile yaml file to disk as NewClientWithCreds doesn't do that by itself.
 	tc.SaveProfile(false /* makeCurrent */)
```

Comment:
```suggestion
	err = tc.SaveProfile(false /* makeCurrent */)
	require.NoError(t, err)
```

I forgot about checking `err` here when I was writing this test.